cirq-google/cirq_google/ops/fsim_gate_family.py

```diff
@@ -0,0 +1,369 @@
+    >>> assert gf.convert(cirq.PhasedFSimGate(theta, 0, 0, 0, np.pi), cirq.CZPowGate) == cirq.CZ
+    >>> assert gf.convert(cirq.FSimGate(theta, phi), cirq.IdentityGate) == cirq.IdentityGate(2)
+
+    3. To accept instances of `gate_types_to_check` based on type / value equality check against
+    gates in `gates_to_accept`, possibly accepting parameterized instances (assuming correct
+    parameter value would be filled in during parameter resolution) based on `allow_symbols`:
+    >>> gf = cirq_google.FSimGateFamily(
+    >>>     gates_to_accept=[cirq.SQRT_ISWAP, cirq.CZPowGate, cirq_google.SYC],
+    >>>     gate_types_to_check=[cirq.FSimGate],
+    >>>     allow_symbols=True,
+    >>> )
+    >>> theta, phi = sympy.Symbol("theta"), sympy.Symbol("phi")
+    >>> assert cirq.FSimGate(theta, phi) in gf # Assumes correct theta/phi will be substituted.
+    >>> assert cirq_google.SYC in gf # SYC
+    >>> assert cirq.FSimGate(0, np.pi / 2) in gf # CZPowGate
+    >>> assert cirq.FSimGate(-np.pi / 4, phi) in gf # SQRT_ISWAP
+    >>> assert cirq.FSimGate(-np.pi / 8, phi) not in gf # No value of `phi` would make it equal to
+    >>>                                                 # any gate/gate type in `gates_to_accept`.
+    >>> assert cirq.CZ ** 0.25 not in gf # CZPowGate not in gate_types_to_check
+    >>> assert cirq.SQRT_ISWAP not in gf # ISwapPowGate not in gate_types_to_check
+    """
+
+    def __init__(
+        self,
+        *,
+        gates_to_accept: Sequence[Union[Type[POSSIBLE_FSIM_GATES], POSSIBLE_FSIM_GATES]] = (),
+        gate_types_to_check: Sequence[Type[POSSIBLE_FSIM_GATES]] = (),
+        allow_symbols: bool = False,
+        atol=_DEFAULT_ATOL,
+    ):
+        """Inits `cirq_google.FSimGateFamily`.
+
+        Args:
+            gates_to_accept: List of gate types or instances to be accepted. All elements
+                should be either types from `POSSIBLE_FSIM_GATES` or non-parameterized
+                instances of those types.
+            gate_types_to_check: List of `POSSIBLE_FSIM_GATES` types whose instances should be
+                considered when trying to match against gates present in `gates_to_accept`.
+                Defaults to all `POSSIBLE_FSIM_GATES` if left unspecified.
+            allow_symbols: If True, both the gate conversion logic and containment predicate allow
+                parameterized gate instances and return a converted gate / accept input gate if
+                there exists any value of the unknown parameters which can result in a valid
+                outcome.
+            atol: Absolute tolerance for difference floating point comparisons.
+
+        Raises:
+            ValueError: If any gate in `gates_to_accept` is not a non-parameterized instance of /
+                or a gate type from `POSSIBLE_FSIM_GATES`.
+            ValueError: If any gate type in `gate_types_to_check` is not one of
+                `POSSIBLE_FSIM_GATES`.
+        """
+        self._supported_types: Dict[
```

Comment:
This should be a member variable because it's a dictionary where values are member functions of the class. The `_convert_to_*` need to be member functions instead of free functions because they in turn call `_approx_eq_or_symbol` member function which depends on `self.allow_symbols` variable. 

We could refactor to make `allow_symbols` a function parameter and then move the member functions out of the class but I don't see much value in doing that. LMK if you feel otherwise.